File: backend-lamas/src/main/java/ch/lamas/backendlamas/statistics/StatisticService.java
package ch.lamas.backendlamas.statistics;


import ch.lamas.backendlamas.statistics.models.StatisticResponse;

import java.util.List;

public interface StatisticService {

    List<StatisticResponse> getDelayLine();

    List<StatisticResponse> getDelayStop();

    List<StatisticResponse> getDelayTransporttype();

    List<StatisticResponse> getAverageDelayLine();

    List<StatisticResponse> getAverageDelayStop();

    List<StatisticResponse> getAverageDelayTransporttype();

}

File: frontend-lamas/src/app/service/statistic.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { StatisticResponse } from '../Model/StatisticResponse';


@Injectable({
  providedIn: 'root'
})
export class StatisticService {

  constructor(private http: HttpClient) { }
  private _url: string = "http://localhost:8080/api/v1/statistics"

  getDelayLine() {
    return this.http.get<StatisticResponse[]>(this._url + "/delayLine")
  }

  getDelayStop() {
    return this.http.get<StatisticResponse[]>(this._url + "/delayStop")
  }

  getDelayTransport() {
      return this.http.get<StatisticResponse[]>(this._url + "/delayTransport")
  }

  getAverageDelayLine() {
    return this.http.get<StatisticResponse[]>(this._url + "/averageDelayLine")
  }

  getAverageDelayStop() {
    return this.http.get<StatisticResponse[]>(this._url + "/averageDelayStop")
  }

  getAverageDelayTransport() {
      return this.http.get<StatisticResponse[]>(this._url + "/averageDelayTransport")
  }
}
